lib/autoproj/ops/build.rb

```diff
@@ -84,7 +86,26 @@ def force_build_packages(selected_packages, all_enabled_packages)
             def build_packages(all_enabled_packages, options = Hash.new)
                 Autobuild.do_rebuild = false
                 Autobuild.do_forced_build = false
-                Autobuild.apply(all_enabled_packages, "autoproj-build", ['build'], options)
+                begin
+                    Autobuild.apply(all_enabled_packages, "autoproj-build", ['build'], options)
+                ensure
+                    build_report(all_enabled_packages)
+                end
+            end
+
+            TEMPLATES_DIR = File.join(File.dirname(__dir__), 'templates')
+
+            def build_report(package_list)
+                Autobuild.message "INFO: creating build report...."
+
+                log_dir = Autobuild.logdir
+            
+                Autobuild.register_logfile("build_report.json")
```

Comment:
Make this name a class-level constant, and provide an accessor for the full path to the file, e.g.

~~~
def build_report_path
   File.join(ws.log_dir, BUILD_REPORT_BASENAME)
end
~~~